internal/declcfg/load.go

```diff
@@ -71,32 +71,32 @@ func extractCSV(objs []string) string {
 	return ""
 }
 
-func readJSON(r io.Reader) (*DeclarativeConfig, error) {
+func readYAMLOrJSON(r io.Reader) (*DeclarativeConfig, error) {
 	cfg := &DeclarativeConfig{}
-	dec := json.NewDecoder(r)
-	for dec.More() {
-		doc := &json.RawMessage{}
-		if err := dec.Decode(doc); err != nil {
-			return cfg, nil
+	dec := yaml.NewYAMLOrJSONDecoder(r, 4096)
```

Comment:
One quick comment here as a note that this `NewYAMLOrJSONDecoder` doesn't validate the JSON blob if it is a string in the value of a key-value. Given most of the value is `json.RawMessage`, this shouldn't be an issue. Plus, there is a test to check for this in the codebase. Consider it handled.